spec/models/spree/invoice_spec.rb

```diff
@@ -0,0 +1,30 @@
+RSpec.describe Spree::Order do
+  subject { Spree::Invoice.new }
+
+  describe 'attributes' do
+    it { is_expected.to respond_to(:number) }
+    it { is_expected.to respond_to(:date) }
+    it { is_expected.to respond_to(:invoiceable) }
+  end
+
+  describe 'creation' do
+    let(:order) { create(:invoiceable_order) }
+
+    it "creates an order on the fly" do
+      expect { order }.to change(Spree::Invoice, :count).by(1)
+    end
+
+    it "automatically has an invoice number" do
+      order
```

Comment:
Since you set the date and number values `after_initialize` you could spare the order creation and simply instantiate a `Spree::Invoice`, right?